modules/local/merge_predictions.nf

```diff
@@ -0,0 +1,54 @@
+process MERGE_PREDICTIONS {
+    label 'process_single'
+    tag "${metadata.sample}"
+
+    conda "bioconda::mhcgnomes=1.8.4"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mhcgnomes:1.8.4--pyh7cba7a3_0' :
+        'quay.io/biocontainers/mhcgnomes:1.8.4--pyh7cba7a3_0' }"
+
+    input:
+    tuple val(metadata), path(prediction_files), path(metadata_file)
+
+    output:
+    tuple val(metadata), path("*.tsv"), emit: merged
+    path "versions.yml", emit: versions
+
+    script:
+    def output = prediction_files.first().baseName.split("_").dropRight(2).join("_")
+    def min_length = (metadata.mhc_class == "I") ? params.min_peptide_length_mhc_I : params.min_peptide_length_mhc_II
```

Comment:
I think it makes sense to move the length filter already to the prepare module since we don't want to predict peptides we would filter out here